pkg/inject/inject.go

```diff
@@ -427,262 +432,85 @@ func (conf *ResourceConfig) injectPodSpec(patch *Patch) {
-			},
-			{
-				Name:  envDestinationSvcAddr,
-				Value: conf.proxyDestinationAddr(),
-			},
-			{
-				Name:  envControlListenAddr,
-				Value: conf.proxyControlListenAddr(),
-			},
-			{
-				Name:  envAdminListenAddr,
-				Value: conf.proxyAdminListenAddr(),
-			},
-			{
-				Name:  envOutboundListenAddr,
-				Value: conf.proxyOutboundListenAddr(),
-			},
-			{
-				Name:  envInboundListenAddr,
-				Value: conf.proxyInboundListenAddr(),
-			},
-			{
-				Name:  envDestinationProfileSuffixes,
-				Value: conf.proxyDestinationProfileSuffixes(),
-			},
-			{
-				Name:  envInboundAcceptKeepAlive,
-				Value: fmt.Sprintf("%dms", defaultKeepaliveMs),
-			},
-			{
-				Name:  envOutboundConnectKeepAlive,
-				Value: fmt.Sprintf("%dms", defaultKeepaliveMs),
-			},
-			{
-				Name:      "_pod_ns",
-				ValueFrom: &corev1.EnvVarSource{FieldRef: &corev1.ObjectFieldSelector{FieldPath: "metadata.namespace"}},
-			},
-			{
-				Name:  envDestinationContext,
-				Value: "ns:$(_pod_ns)",
-			},
-		},
-		ReadinessProbe: conf.proxyReadinessProbe(),
-		LivenessProbe:  conf.proxyLivenessProbe(),
-	}
+	saVolumeMount := conf.serviceAccountVolumeMount()
 
 	// use the primary container's capabilities to ensure psp compliance, if
 	// enabled
 	if conf.pod.spec.Containers != nil && len(conf.pod.spec.Containers) > 0 {
-		if securityContext := conf.pod.spec.Containers[0].SecurityContext; securityContext != nil {
-			sidecar.SecurityContext.Capabilities = securityContext.Capabilities
-		}
-	}
-
-	// Special case if the caller specifies that
-	// LINKERD2_PROXY_OUTBOUND_ROUTER_CAPACITY be set on the pod.
-	// We key off of any container image in the pod. Ideally we would instead key
-	// off of something at the top-level of the PodSpec, but there is nothing
-	// easily identifiable at that level.
-	// Currently this will be set on any proxy that gets injected into a Prometheus pod,
-	// not just the one in Linkerd's Control Plane.
-	for _, container := range conf.pod.spec.Containers {
-		if capacity, ok := conf.proxyOutboundCapacity[container.Image]; ok {
-			sidecar.Env = append(sidecar.Env,
-				corev1.EnvVar{
-					Name:  "LINKERD2_PROXY_OUTBOUND_ROUTER_CAPACITY",
-					Value: fmt.Sprintf("%d", capacity),
-				},
-			)
-			break
+		if sc := conf.pod.spec.Containers[0].SecurityContext; sc != nil && sc.Capabilities != nil {
+			values.Proxy.Capabilities = &charts.Capabilities{
+				Add:  []string{},
+				Drop: []string{},
+			}
+			for _, add := range sc.Capabilities.Add {
+				values.Proxy.Capabilities.Add = append(values.Proxy.Capabilities.Add, string(add))
+			}
+			for _, drop := range sc.Capabilities.Drop {
+				values.Proxy.Capabilities.Drop = append(values.Proxy.Capabilities.Drop, string(drop))
+			}
 		}
 	}
 
-	if conf.tapDisabled() {
-		sidecar.Env = append(sidecar.Env,
-			corev1.EnvVar{
-				Name:  envTapDisabled,
-				Value: "true",
-			},
-		)
+	if !conf.configs.GetGlobal().GetCniEnabled() {
+		conf.injectProxyInit(values)
```

Comment:
From what I'm seeing in my test, the `SAMountPath` isn't set in the `proxy-init`. This call to `injectProxyInit()` needs to happen after line 472 below.

Reply:
Good catch :+1: